### PR TITLE
some elements changed to dark mode

### PR DESCRIPTION
**Pull request by IsaacHatton**
Changing some elements to make them work with dark mode, definitely not finished in all elements as I don't have a crossway key so I can't load up the entire app.

### CODE REVIEW

**Comment by IsaacHatton** [comment]
Thanks for making the vercel deploy preview work, I will try and make the remaining required changes by referencing this.

**Comment by joshuawootonn** [comment]
Hey @IsaacHatton. Thanks for putting this together! If you rebase onto main I just updated the project to load Genesis 1, Psalm 23, and James 1 without and API key. Hope that is helpful!

**Comment by IsaacHatton** [comment]
Thanks, that's really helpful! 
Will finish making the rest of the app functional in dark mode now.

**Comment by joshuawootonn** [comment]
Sweet!

**Comment by joshuawootonn** [comment]
Hey also, I like the idea of having this based solely on system preferences. So no need to add a toggle button!

**Comment by IsaacHatton** [comment]
Hi,
I have made all the changes that I have spotted, please can you check as I may have missed one somewhere (also I haven't tested what happens when a user logs in as the menu changes and the history page because I haven't generated myself a development Google OAuth ID/Secret)

File: src/styles/globals.css [hunk 1]
```diff
@@ -28,4 +28,19 @@ body[style] {
...
+/* if dark mode make scrollbar dark */
+@media (prefers-color-scheme: dark) {
+    ::-webkit-scrollbar {
+        @apply bg-black;
+    }
+    ::-webkit-scrollbar-thumb {
+        @apply bg-gray-500;
+        border-radius: 10px;
+    }
 }
```
Comment (joshuawootonn) [inline src/styles/globals.css:58]: Custom scrollbar styles can be unpredicable on mobile browsers and stuff.

After some [digging](https://stackoverflow.com/a/66097469/8239711) though this should do the trick:
```
:root {
    color-scheme: dark;
}
```

File: src/pages/why.tsx [hunk 2]
```diff
@@ -93,13 +93,13 @@ export default function Home() {
...
                     <hr className="mx-0 w-full border-t-2  border-black" />
-                    <h2>Why meditate on God&apos;s word?</h2>
+                    <h2 className="dark:text-white">Why meditate on God&apos;s word?</h2>
                     <p>
                         In the process of making typetheword.site, these
                         scriptures were great reminders of why we should
                         meditate on God&apos;s word. If you have others, feel
                         free to{' '}
-                        <EmailLink className="underline">email me</EmailLink>. I
+                        <EmailLink className="underline dark:text-white">email me</EmailLink>. I
```
Comment (joshuawootonn) [inline src/pages/why.tsx]: I think this change will be used in all usages of `EmailLink` so I would prefer to have this within the component text itself.

File: src/components/emailLink.tsx [hunk 3]
```diff
@@ -56,7 +56,7 @@ export function EmailLink({
             >
                 <div
                     className={clsx(
-                        'border-2 border-black bg-white px-2 py-1 font-medium',
+                        'border-2 border-black bg-white dark:bg-black px-2 py-1 font-medium',
```
Comment (joshuawootonn) [inline src/components/emailLink.tsx]: I would like this to also be `dark:border-white`

File: package-lock.json
Comment (joshuawootonn) [inline package-lock.json:1]: This should be deleted. I am using `pnpm` for the package management.

File: src/components/arena.tsx [hunk 4]
```diff
@@ -86,14 +86,14 @@ export function Arena({
 
                                 case 'h2':
                                     return (
-                                        <h2 key={pIndex} className="text-2xl">
+                                        <h2 key={pIndex} className="text-2xl dark:text-white">
```
Comment (joshuawootonn) [inline src/components/arena.tsx:89]: Same comment on this being styled with tailwind typography.

File: src/components/currentVerse.tsx [hunk 5]
```diff
@@ -303,7 +303,7 @@ export function CurrentVerse({
     return (
         <span
             className={clsx(
-                'verse break-spaces text-balance group inline h-3 hover:cursor-pointer',
+                'verse break-spaces text-balance group inline h-3 hover:cursor-pointer dark:text-white',
```
Comment (joshuawootonn) [inline src/components/currentVerse.tsx:306]: These can likely be removed by doing `dark:prose-p:text-white`.

File: src/pages/index.tsx [hunk 6]
```diff
@@ -52,7 +52,7 @@ export default function Home(props: { passage?: PassageReference }) {
     const router = useRouter()
 
     return (
-        <div className="min-h-screen-1px container mx-auto flex max-w-page flex-col px-4 lg:px-0">
+        <div className="dark:bg-black min-h-screen-1px container mx-auto flex max-w-page flex-col px-4 lg:px-0">
```
Comment (joshuawootonn) [inline src/pages/index.tsx:55]: Is this still required with the?
```
body {
    @apply dark:bg-black;
}
```

**Review by joshuawootonn — COMMENTED** [review]
Hey! this is looking great. I was worried that the white on black would be to strong, but I am loving it.

I have two main comments:
- I would prefer to not add custom styles to the scroll bar and instead user `color-scheme: dark`
- I would prefer to keep text color changes in tailwind typography as much as we can. You can tell where I am using tailwind typography by the `prose` class.

Overall though this is going great. Thanks for jumping in 🙌🏼

**Comment by joshuawootonn** [comment]
> because I haven't generated myself a development Google OAuth ID/Secret)

Is this just because you didn't want to login with google? Or did you run into something weird when logging in?

I am asking because the app is in the middle of verification right now.

**Comment by IsaacHatton** [comment]
> > because I haven't generated myself a development Google OAuth ID/Secret)
> 
> Is this just because you didn't want to login with google? Or did you run into something weird when logging in?
> 
> I am asking because the app is in the middle of verification right now.

When I tried to login on the deploy preview, the url of the deploy preview didn't match the website so it refused.
When I am running it locally, I just put 0 as my environment variable for ID/Secret, as I don't have anything to put there.
When I login on the main site, it works fine.

**Comment by IsaacHatton** [comment]
Hi @joshuawootonn,

I have tried to make some changes, please can you check if they are working properly.
In the arena, it still seems to need dark:text-white for some reason, I know its a mistake I am making but I can't see where it is.

**Comment by joshuawootonn** [comment]
@IsaacHatton I went through these files line by line to see what was needed and not.
CleanShot 2023-12-19 at 07 07 11@2x  

I did this because I want as much of the dark mode styles to come from inherited colors as possible. In theory, it would be great if we could just set `dark:text-white` on the body element and just take care of the few exceptions to that, and just let all elements inherit that color. 

The reason you were running into a problem in `arena.tsx` was because you had `dark:invert-prose` instead of `dark:prose-invert`. Happens to the best of us :)

If you could go through the remaining files and double check that only the critical things have changed I would appreciate it. 

If you can do that I can resolve the conflict and get this merged tomorrrow.

**Comment by IsaacHatton** [comment]
Ok, I will go through them shortly.

**Comment by IsaacHatton** [comment]
I have gone through the files, think I have caught all of them, but I may have missed some.

**Comment by joshuawootonn** [comment]
Merged! Thanks for the help @IsaacHatton 🙌🏼

**Comment by joshuawootonn** [comment]
Yo @IsaacHatton. I have never had anyone contribute to a project I created. Just wanted to say that this is kinda big for me. I actually do really appreciate your initiative. 

I would love to meet you. If you are free in January, let chat: https://calendly.com/joshuawootonn/1on1

No real expectation here, I just like meeting other developers and hearing their story.

**Pull request merged**